src/internal_handlers/peer_management/tester.rs

```diff
@@ -9,12 +9,17 @@ use crate::{
     handlers::{MessageHandler, MessageHandlers},
     internal_handlers::peer_management::{announcement::Announcement, PeerInfo},
     network_manager::PeerNetManager,
+    peer::HandshakeHandler,
     peer_id::PeerId,
     transports::{endpoint::Endpoint, OutConnectionConfig, TcpOutConnectionConfig, TransportType},
 };
 
 use super::{PeerDB, SharedPeerDB};
 
+#[derive(Clone)]
+pub struct EmptyHandshake;
+impl HandshakeHandler for EmptyHandshake {}
```

Comment:
For the tester it need to have access to peerdb.
- Receive the announcement. (deserialize)
- Add to peerdb if valid